### PR TITLE
Config value named metadata to be part of payload

### PR DESCRIPTION
**Pull request by goutkannan**
The field `metadata` from config will be shoveled into the payload as `metadata` filed. 

@joshperry  @film42  @jonphilpott

### CODE REVIEW

File: spec/lib/logstasher/log_subscriber_spec.rb [hunk 1]
```diff
@@ -10,15 +10,22 @@ def user_id
...
+    @env = { 'action_dispatch.request_id' => 1 }
+  end
+
   def remote_ip
     '127.0.0.1'
   end
 end
 
-describe LogStasher::LogSubscriber do
+ describe LogStasher::LogSubscriber do
```
Comment (film42) [inline spec/lib/logstasher/log_subscriber_spec.rb:24]: Extra space here?

Comment (goutkannan) [reply spec/lib/logstasher/log_subscriber_spec.rb:24]: :( I will remove it.

File: lib/logstasher/log_subscriber.rb [hunk 2]
```diff
@@ -62,6 +61,11 @@ def extract_request(payload)
         :path       => extract_path(payload),
         :route      => "#{payload[:controller]}##{payload[:action]}"
       }
+      metadata = ::LogStasher.metadata
+      if !!metadata
```
Comment (film42) [inline lib/logstasher/log_subscriber.rb:65]: This will always be true because of `config.logstasher.metadata  = {}` in `railtie.rb`.

`!!{} #=> true`

Comment (film42) [reply lib/logstasher/log_subscriber.rb:65]: You can also say:
```
result.merge!(:metadata => metadata) unless metadata&.empty?
```

**Review by film42 — APPROVED** [review]

**Pull request merged**